_rules/iframe-accessible-name-cae760.md

```diff
@@ -21,23 +21,23 @@ acknowledgments:
 
 ## Applicability
 
-The rule applies to `iframe` elements that are [included in the accessibility tree][] or that are part of [sequential focus navigation][].
+The rule applies to `iframe` elements that are [included in the accessibility tree][] and has contents that can be accessed by [sequential focus navigation][].
```

Comment:
I'm not clear why "has contents" is included here. Is it only applicable for non-empty iframes? I think the original text is better. Line 40 with the update seems to cover iframe content better.

Reply:
Adding the following AT & browser combos:

* JAWS & Chrome.
* NVDA & Firefox.

| scenario | VO result | Chrome & JAWS result |  Firefox & NVDA result |
| --- | --- | --- | --- |
| empty frame | VO ignores it completely | Chrome & JAWS do not speak it, although there seems to be a tab stop for the frame | Firefox and NVDA announce iframe title |
| empty iframe with tabindex=1 | VO ignores it completely | Chrome & JAWS do not speak it, although there seems to be a tab stop for the frame | Firefox and NVDA announce iframe title |
| iframe with focusable content | VO announces the iframe | Chrome & JAWS announce the iframe title and contents as you tab to the iframe | Firefox and NVDA announce the iframe title and contents as you tab to the iframe |
| iframe with non-focusable content | VO announces the iframe | JAWS & Chrome announce the iframe title, iframe content needs to be reviewed using browse mode | Firefox and NVDA announce the iframe title and contents as you tab to the iframe |

> Interesting caveat; VO in Safari does not look at aria-label. It only announces title. We should put that in Accessibility support.

Both JAWS and NVDA work well with `title` and `aria-label`
> 
> This tells me what matters here, at least for VO is if the iframe has any content. I tested with an image with alt="", if that's the only thing in the frame, VO ignores it. I think we need to do more testing before we continue with this one.

I may need some visual help in Chrome. As said above, there seems to be tab stops for each of the empty iframes (with and without tabindex value), but I am not sure if this is really happening because there is a frame or because there is a title. When I remove titles from empty iframes, tab stops seem to disappear.

Reply:
> This one keeps getting more complicated... :( I think we made a mistake with the "is in sequential focus navigation". I did the following tests with Voiceover:
> 
> scenario	VO result
> empty frame	VO ignores it completely
> empty iframe with tabindex=1	VO ignores it completely
> iframe with focusable content	VO announces the iframe
> iframe with non-focusable content	VO announces the iframe
> Interesting caveat; VO in Safari does not look at aria-label. It only announces title. We should put that in Accessibility support.
> 
> This tells me what matters here, at least for VO is if the iframe has any content. I tested with an image with alt="", if that's the only thing in the frame, VO ignores it. I think we need to do more testing before we continue with this one.

Adding the following AT & browser combos:

* JAWS & Chrome.
* NVDA & Firefox.

| scenario | VO result | Chrome & JAWS result |  Firefox & NVDA result |
| --- | --- | --- | --- |
| empty frame | VO ignores it completely | Chrome & JAWS do not speak it, although there seems to be a tab stop for the frame | Firefox and NVDA announce iframe title |
| empty iframe with tabindex=1 | VO ignores it completely | Chrome & JAWS do not speak it, although there seems to be a tab stop for the frame | Firefox and NVDA announce iframe title |
| iframe with focusable content | VO announces the iframe | Chrome & JAWS announce the iframe title and contents as you tab to the iframe | Firefox and NVDA announce the iframe title and contents as you tab to the iframe |
| iframe with non-focusable content | VO announces the iframe | JAWS & Chrome announce the iframe title, iframe content needs to be reviewed using browse mode | Firefox and NVDA announce the iframe title and contents as you tab to the iframe |

> Interesting caveat; VO in Safari does not look at aria-label. It only announces title. We should put that in Accessibility support.

Both JAWS and NVDA work well with `title` and `aria-label`
> 
> This tells me what matters here, at least for VO is if the iframe has any content. I tested with an image with alt="", if that's the only thing in the frame, VO ignores it. I think we need to do more testing before we continue with this one.

As said above, in Chrome there seems to be tab stops for each of the empty iframes (with and without tabindex value), there is no visual clue that those empty frames are focused but there might be some programmatic focus there. This might relate to https://github.com/act-rules/act-rules.github.io/issues/1170#issue-564753539